src/OneDriveAuth.js

```diff
@@ -289,10 +289,15 @@ export class OneDriveAuth {
       } catch (e) {
         this.handleAcquireError(account, e);
       }
+    }
+
+    // try again with fresh mem cache
+    if (this.cachePlugin instanceof MemCachePlugin) {
+      this.cachePlugin.clear();
 
-      // try again with fresh mem cache
-      if (this.cachePlugin instanceof MemCachePlugin) {
-        this.cachePlugin.clear();
+      accounts = await app.getTokenCache().getAllAccounts();
```

Comment:
This call actually calls the plugin's `beforeCacheAccess` again. Because we clear the mem cache, it will invoke the base plugin's `beforeCacheAccess`